R/memprof.R

```diff
@@ -1,3 +1,172 @@
-add <- function(a, b) {
-  a + b
+#' Run an R expression and monitor memory
+#'
+#' Evaluate an expression, and monitor the memory every `poll_interval`.
+#' This will monitor the memory of your entire system, not just the expression
+#' being run.
+#'
+#' @param expr The expression to evaluate.
+#' @param poll_interval Time between each check of memory use.
+#' @param monitor_file Path to log memory monitor to, you can use this if
+#'   your expr errors to recover the output for later plotting. Uses a
+#'   tempfile by default.
+#' @param overwrite If TRUE then overwrite `monitor_file` if it exists already.
+#' @param gc If `TRUE` run a garbage collect before starting monitoring.
+#'
+#' @return A `memprof_monitor` object containing the `result` as the outcome
+#'   from the expression and `memory_use` the memory used
+#' @export
+with_monitor <- function(expr, poll_interval = 0.1,
+                         monitor_file = tempfile(),
+                         overwrite = FALSE,
+                         gc = TRUE) {
+  validate_monitor_file(monitor_file, overwrite)
+  memory <- monitor$new(monitor_file, poll_interval, gc)
+  on.exit(memory$stop())
+  result <- force(expr)
+  used <- memory$finish()
+  build_memprof_monitor(result, used)
+}
+
+
+#' Build `memprof_monitor` object from log file which can be used for plotting.
+#'
+#' @param log_file Path to memprof log file.
+#'
+#' @return A `memprof_monitor` object.
+#' @export
+monitor_from_log <- function(log_file) {
```

Comment:
Argument names here are inconsistent between `log_file` and `monitor_file`

I also wonder if this is better as `monitor_read` as you're not really constructing a `monitor` object (even though we don't yet expose that to the user)